src/views/smart_table.rs

```diff
@@ -0,0 +1,219 @@
+use vizia::prelude::*;
+
+pub const CELL_MIN_SIZE_PX: f32 = 100.0;
+
+#[derive(Clone, Copy, Debug)]
+pub enum SmartTableEvent {
+    StartDrag(usize),
+    StopDrag,
+}
+
+#[derive(Lens)]
+pub struct SmartTable {
+    dragging: Option<usize>,
+    initialized: bool,
+    limiters: Vec<f32>,
+    sizes: Vec<f32>, // derived
+}
+
+impl<'a> SmartTable {
+    pub fn new<S>(cx: &'a mut Context, mut rows: Vec<Vec<S>>) -> Handle<'a, Self>
+    where
+        S: Into<&'a str> + Clone,
+    {
+        let transformed: Vec<Vec<&str>> =
+            rows.iter_mut().map(|row| row.iter().map(|d| d.clone().into()).collect()).collect();
+
+        drop(rows);
```

Comment:
Why explicitly drop here? Won't it get dropped at the end of this function anyway?

Reply:
Yes, but there was some bug that was confusing me while developing the Smart Table and I completely forgot to get rid of the `drop` function. I'll change it ASAP